### PR TITLE
Add unit tests of the PlatformBridgeCertValidator

### PR DESCRIPTION
**Pull request by RyanTheOptimist**
Risk Level: Low
Testing: Yes :)
Docs Changes: N/A
Release Notes: N/A

### CODE REVIEW

**Comment by RyanTheOptimist** [comment]
@alyssawilk PTAL

File: test/common/extensions/cert_validator/platform_bridge/platform_bridge_cert_validator_test.cc [hunk 1]
```diff
@@ -0,0 +1,314 @@
...
+
+  bool waitForDispatcherToExit() {
+    bool timer_fired = false;
+    Event::TimerPtr timer(dispatcher_->createTimer([this, &timer_fired]() -> void {
+      timer_fired = true;
+      dispatcher_->exit();
+    }));
+    timer->enableTimer(std::chrono::milliseconds(100));
+    dispatcher_->run(Event::Dispatcher::RunType::RunUntilExit);
+    return timer_fired;
```
Comment (alyssawilk) [inline test/common/extensions/cert_validator/platform_bridge/platform_bridge_cert_validator_test.cc:90]: can you skip the bool and return timer->enabled()?

File: test/common/extensions/cert_validator/platform_bridge/platform_bridge_cert_validator_test.cc [hunk 2]
```diff
@@ -0,0 +1,314 @@
...
+    EXPECT_CALL(config_, certificateRevocationList()).WillOnce(ReturnRef(empty_string_));
+    EXPECT_CALL(config_, trustChainVerification()).WillOnce(Return(GetParam()));
+  }
+
+  ~PlatformBridgeCertValidatorTest() {
+    mock_validator_.reset();
+    main_thread_id_ = std::thread::id();
+  }
+
+  bool waitForDispatcherToExit() {
```
Comment (alyssawilk) [inline test/common/extensions/cert_validator/platform_bridge/platform_bridge_cert_validator_test.cc:82]: optional: ABSL_MUST_USE_RESULT?

File: test/common/extensions/cert_validator/platform_bridge/platform_bridge_cert_validator_test.cc [hunk 3]
```diff
@@ -0,0 +1,314 @@
...
+    return timer_fired;
+  }
+
+  bool acceptInvalidCertificates() {
+    return GetParam() == CertificateValidationContext::ACCEPT_UNTRUSTED;
+  }
+
+  static envoy_cert_validation_result validate(const envoy_data* certs, uint8_t size,
+                                               const char* hostname) {
+    // Validate must be called on the worked thread, not the main thread.
```
Comment (alyssawilk) [inline test/common/extensions/cert_validator/platform_bridge/platform_bridge_cert_validator_test.cc:99]: worker (here and below)

File: test/common/extensions/cert_validator/platform_bridge/platform_bridge_cert_validator_test.cc [hunk 4]
```diff
@@ -0,0 +1,314 @@
...
+
+TEST_P(PlatformBridgeCertValidatorTest, NonEmptyRevocationList) {
+  std::string revocation_list = "xyz";
+  EXPECT_CALL(config_, caCert()).WillRepeatedly(ReturnRef(empty_string_));
+  EXPECT_CALL(config_, certificateRevocationList()).WillRepeatedly(ReturnRef(revocation_list));
+  EXPECT_CALL(config_, trustChainVerification()).WillRepeatedly(Return(GetParam()));
+
+  EXPECT_ENVOY_BUG(
+      { PlatformBridgeCertValidator validator(&config_, stats_, &platform_validator_); },
+      "Invalid certificate validation context config.");
```
Comment (alyssawilk) [inline test/common/extensions/cert_validator/platform_bridge/platform_bridge_cert_validator_test.cc:165]: are there details in the log which would let us differentiate this invalid from the above invalid?

**Review by alyssawilk — COMMENTED** [review]
Yay!  Unit tests FTW!
/wait

**Review by RyanTheOptimist — COMMENTED** [review]
Thanks!

**Review by alyssawilk — APPROVED** [review]

**Review by alyssawilk — DISMISSED** [review]

File: test/common/extensions/cert_validator/platform_bridge/platform_bridge_cert_validator_test.cc [hunk 5]
```diff
@@ -77,6 +78,7 @@ class PlatformBridgeCertValidatorTest
   ~PlatformBridgeCertValidatorTest() {
     mock_validator_.reset();
     main_thread_id_ = std::thread::id();
+    Envoy::Assert::resetEnvoyBugCountersForTest();
```
Comment (alyssawilk) [inline test/common/extensions/cert_validator/platform_bridge/platform_bridge_cert_validator_test.cc:81]: Wow - how frustrating.  Glad you found it!

**Review by alyssawilk — APPROVED** [review]

**Pull request merged**